webpack/index.d.ts

```diff
@@ -689,6 +689,84 @@ declare namespace webpack {
         constructor(definitions: {[key: string]: any});
     }
 
+    class DllPlugin extends Plugin {
+        constructor(options: DllPlugin.Options | DllPlugin.Options[]);
+    }
+
+    namespace DllPlugin {
+        interface Options {
+            /**
+             * The absolute path to the manifest json file (output).
+             */
+            path: string,
```

Comment:
Reformatted and re-organized type members.

Added tests as well. Can you review these? The gist of the test file as far as I could tell was to verify a usage of type declarations, but not to test against the actual Webpack implementation. I tried to add the tests in that format.